sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClient.cs

```diff
@@ -1355,8 +1358,15 @@ private void Initialize(CancellationToken cancellationToken)
             try
             {
                 Response<Key> key = _remoteProvider.GetKey(cancellationToken);
-                _provider = LocalCryptographyProviderFactory.Create(key.Value.KeyMaterial);
-                ;
+
+                _provider = LocalCryptographyProviderFactory.Create(key.Value);
+                if (_provider is null)
+                {
+                    // TODO: Log that the key type is unsupported locally.
+
+                    _provider = _remoteProvider;
+                    return;
+                }
             }
             catch (RequestFailedException e) when (e.Status == 403)
```

Comment:
I would prefer that we handle the status code directly rather letting the exception being thrown. I don't think that this needs to be fixed for this PR but we should create an issue to update this both here and in KeyResolver

Reply:
Opened #8040